validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorClientService.java

```diff
@@ -235,23 +245,39 @@ public static ValidatorClientService create(
   private void initializeDoppelgangerDetector(
       AsyncRunner asyncRunner,
       ValidatorApiChannel validatorApiChannel,
-      ValidatorIndexProvider validatorIndexProvider,
       Spec spec,
       TimeProvider timeProvider,
       GenesisDataProvider genesisDataProvider) {
     DoppelgangerDetector doppelgangerDetector =
         new DoppelgangerDetector(
             asyncRunner,
             validatorApiChannel,
-            validatorIndexProvider,
             spec,
             timeProvider,
             genesisDataProvider,
             DOPPELGANGER_DETECTOR_CHECK_DELAY,
-            DOPPELGANGER_DETECTOR_TIMEOUT);
+            DOPPELGANGER_DETECTOR_TIMEOUT,
+            DOPPELGANGER_DETECTOR_MAX_EPOCHS);
     maybeDoppelgangerDetector = Optional.of(doppelgangerDetector);
   }
 
+  private void initializeValidatorRestApi(
+      ValidatorRestApiConfig validatorRestApiConfig,
+      Optional<ProposerConfigManager> proposerConfigManager,
+      ActiveKeyManager activeKeyManager,
+      DataDirLayout dataDirLayout,
+      Optional<DoppelgangerDetector> maybeDoppelgangerDetector) {
+    RestApi validatorRestApi =
+        ValidatorRestApi.create(
+            validatorRestApiConfig,
+            proposerConfigManager,
+            activeKeyManager,
+            dataDirLayout,
+            maybeDoppelgangerDetector,
+            Optional.of(new DoppelgangerDetectionAlert()));
```

Comment:
So for these `DoppelgangerDetector` I wouldn't use an `Optional` but use pass the class directly. In a test we can easily create a mock implementation to avoid actually shutting down and then the code doesn't have to deal with it potentially not being present.